gcore/gdalrasterband.cpp

```diff
@@ -6580,175 +6612,117 @@ static int
 ComputeStatisticsFloat32_SSE2(const float *pafData,
                               [[maybe_unused]] float fNoDataValue, int iX,
                               int nCount, float &fMin, float &fMax,
-                              float &fBlockMean, float &fBlockM2,
-                              int &nBlockValidCount)
+                              double &dfBlockMean, double &dfBlockM2,
+                              double &dfBlockValidCount)
```

Comment:
Why is this now a floating point type?

Reply:
In the computation, we do a division on type double by the number of valid pixels processed, so it is more efficient to store it on double rather than doing multiple int <--> double conversions.   This is the same as the very similar Float64 implementation of this method.